src/utils/image_saver.py

```
@@ -0,0 +1,9 @@
+import os
+
+def save_image(image, output_path):
+    try:
+        os.makedirs(os.path.dirname(output_path), exist_ok=True)
+        image.save(output_path)
+        print(f"File saved: {output_path}")
+    except Exception as e:
+        raise IOError(f"Error while saving file: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)